### PR TITLE
Added token to options

### PR DESCRIPTION
**Pull request by MikeOne**
Added the possibility to pass in the flex token manually. This allows
you to get the plex token from another source (for example, from pin
auth).

### CODE REVIEW

File: lib/api.js [hunk 1]
```diff
@@ -122,7 +123,7 @@ PlexAPI.prototype._request = function _request(relativeUrl, method, parseRespons
             );
         }
 
-        if (response.statusCode !== 200) {
+        if (response.statusCode !== 200 && response.statusCode !== 201) {
```
Comment (OverloadUT) [inline lib/api.js]: Perhaps we should simply be checking for the entire 2XX response family?

Comment (MikeOne) [inline lib/api.js]: I guess we could for future proofing?

Comment (phillipj) [inline lib/api.js]: 👍

**Comment by phillipj** [comment]
Nice @MikeOne! Any thoughta about the need to document this behavior with a test or two?

File: lib/api.js (hunk 1 above)
Comment (MikeOne) [inline lib/api.js]: So I should add the whole 2xx range @phillipj ?

Comment (phillipj) [inline lib/api.js]: > So I should add the whole 2xx range @phillipj ?

Yupp, havent thought of any scenario where that would be a bad thing.

**Comment by MikeOne** [comment]
I think I might have screwed that up..
Working - yes! but the PR's are a bit confusing now.. Sorry!

File: lib/api.js [hunk 2]
```diff
@@ -122,8 +123,8 @@ PlexAPI.prototype._request = function _request(relativeUrl, method, parseRespons
             );
         }
 
-        if (response.statusCode !== 200) {
-            return deferred.reject(new Error('Plex Server didnt respond with status code 200, response code: ' + response.statusCode));
+        if (response.statusCode < 200 && response.statusCode > 299) {
```
Comment (phillipj) [inline lib/api.js]: Found what's probably causing the test to fail, `statusCode` cannot be both below 200 **and** over 299.

Replacing `&&` with `||` seems to fix it :smile:

**Comment by phillipj** [comment]
About removing unnecessary commits from the PR, you could rebase and force push afterwards.

As of writing this you have 4 commits for these changes, which could be changed into one by following these steps:

```bash
$ git rebase -i fdb8d0f
```
That brings up an editor listing your commits since the last one we've got in the official repo. The 3 latest commits can be marked as `fixup` instead of `pick` meaning they will all be merged into the top commit (92f47eb).

```
pick 92f47eb Added token to options
pick 66c9b6c Added additional 201 status code as valid
pick 3f14549 Allowed status codes 2XX
pick 392d9f6 Corrected code style issue
```
Should end up looking like this:
```
pick 92f47eb Added token to options
fixup 66c9b6c Added additional 201 status code as valid
fixup 3f14549 Allowed status codes 2XX
fixup 392d9f6 Corrected code style issue
```
Then close the editor and it'll do some nifty tricks in the background, resulting in you only having one commit left with all your changes. Verify it by looking at the entries in your log with
```bash
$ git log
```
Even check the changelog of the last commit:
```bash
$ git show
```
If everything seems well, you have to force push it back to github as you have re-written the history of your repo.
```bash
$ git push -f origin master
```
Obligatory disclaimer; force pushing master isnt something you should make a habit of doing! ..although it's necessary in this case as you havent created a feature branch for these changes :smiley:

**Comment by MikeOne** [comment]
I still learn something every day, thanks philip!

File: lib/api.js (hunk 2 above)
Comment (MikeOne) [inline lib/api.js]: Oh my... What was I thinking.. I shouldn't make changes when I'm in a hurry I guess.. Pfff.

**Comment by MikeOne** [comment]
Is it not easier at this point to discard my changes (reject the PR?). Then I'll (re)fork your repo again, make a feature branch containing all fixes, push that and then do a single PR?

**Comment by phillipj** [comment]
Definitely a valid approach :+1: What you consider the simplest solution is entirely up to you :smile:

**Comment by MikeOne** [comment]
Okay, I'll will do it again (and hopefully do it right in one shot :-)). You can reject this PR.

**Comment by MikeOne** [comment]
okay- thanks, to do this right, should I actually do 2 PRs? One for the authToken and one for the allowed status codes? What would you prefer?

**Comment by phillipj** [comment]
Sure, two PRs would be fine. Would also appreciate a test with each of the changes verifying it and ensuring we wont break those expectations later..

**Pull request closed without merging**